### ISSUE TITLE
Variable is lost at dependency role at included tasks

### ISSUE THREAD

**Issue by havlasme** [ansible/ansible#16564]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

I have two roles, one is dependent on another. Inside a playbook I set a variable for the "main" role, this variable is passed to dependency. Dependency tasks are separated into files and then included from `main.yml` using `include`.
If I try to access variable in task at `main.yml` inside dependency role it works as expected, however if I try to use the same variable in tasks located at some included file variable is lost.

NOTE: When the same variable is set under `vars` inside playbook, and not as shown below, everything works as expected.
##### STEPS TO REPRODUCE

`playbook.yml`

```

---
- hosts: localhost
  roles:
    - role: 'dependent'
      t_var: 'value'
```

`roles/dependency/tasks/main.yml`

```

---
- name: debug test_var at main before inc
  debug:
    var: 'test_var'

- include: 'inc.yml'

- name: debug test_var at main after inc
  debug:
    var: 'test_var'
```

`roles/dependency/tasks/inc.yml`

```

---
- name: debug test_var inside inc
  debug:
    var: 'test_var'
```

`roles/dependent/meta/main.yml`

```

---
dependencies:
  - role: 'dependency'
    test_var: '{{ t_var }}'
```

##### EXPECTED RESULTS

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [dependency : debug test_var at main before inc] **************************
ok: [localhost] => {
    "test_var": "value"
}

TASK [dependency : include] ****************************************************
included: .../roles/dependency/tasks/inc.yml for localhost

TASK [dependency : debug test_var inside inc] **********************************
ok: [localhost] => {
    "test_var": "value"
}

TASK [dependency : debug test_var at main after inc] ***************************
ok: [localhost] => {
    "test_var": "value"
}

PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=0
```
##### ACTUAL RESULTS

```
PLAY [localhost] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [dependency : debug test_var at main before inc] **************************
ok: [localhost] => {
    "test_var": "value"
}

TASK [dependency : include] ****************************************************
included: .../roles/dependency/tasks/inc.yml for localhost

TASK [dependency : debug test_var inside inc] **********************************
ok: [localhost] => {
    "test_var": "VARIABLE IS NOT DEFINED!"
}

TASK [dependency : debug test_var at main after inc] ***************************
ok: [localhost] => {
    "test_var": "value"
}

PLAY RECAP *********************************************************************
localhost                  : ok=5    changed=0    unreachable=0    failed=0
```

**Comment by havlasme** [comment]
I have identified problem in file `vars/__init__.py`, caused by using accessing dependency chain by property `_dep_chain` instead of method `get_dep_chain`. I believe that this problem is fixed in development version, so this issue could be closed.

**Comment by alikins** [comment]
Ah, looks like it is fixed in devel/2.2.0. Closing.
